django/core/serializers/python.py

```diff
@@ -75,81 +75,102 @@ def getvalue(self):
+        self.object_list = object_list
+        self.handle_forward_references = options.pop('handle_forward_references', False)
+        self.using = using
+        self.ignorenonexistent = ignorenonexistent
+        self.field_names_cache = {}  # Model: <list of field_names>
+
+    def __iter__(self):
+        for obj in self.object_list:
+            data = {}
+            m2m_data = {}
+            deferred_fields = {}
+            field_names = set()
+
+            # Look up the model and starting build a dict of data for it.
             try:
-                data[Model._meta.pk.attname] = Model._meta.pk.to_python(d.get('pk'))
-            except Exception as e:
-                raise base.DeserializationError.WithData(e, d['model'], d.get('pk'), None)
-        m2m_data = {}
-        deferred_fields = {}
-
-        if Model not in field_names_cache:
-            field_names_cache[Model] = {f.name for f in Model._meta.get_fields()}
-        field_names = field_names_cache[Model]
-
-        # Handle each field
-        for (field_name, field_value) in d["fields"].items():
-
-            if ignorenonexistent and field_name not in field_names:
-                # skip fields no longer on model
-                continue
-
-            field = Model._meta.get_field(field_name)
-
-            # Handle M2M relations
-            if field.remote_field and isinstance(field.remote_field, models.ManyToManyRel):
-                try:
-                    values = base.deserialize_m2m_values(field, field_value, using, handle_forward_references)
-                except base.M2MDeserializationError as e:
-                    raise base.DeserializationError.WithData(e.original_exc, d['model'], d.get('pk'), e.pk)
-                if values == base.DEFER_FIELD:
-                    deferred_fields[field] = field_value
-                else:
-                    m2m_data[field.name] = values
-            # Handle FK fields
-            elif field.remote_field and isinstance(field.remote_field, models.ManyToOneRel):
+                Model = self._get_model(obj["model"])
+            except base.DeserializationError:
+                if self.ignorenonexistent:
+                    continue
+                raise
+            if 'pk' in obj:
                 try:
-                    value = base.deserialize_fk_value(field, field_value, using, handle_forward_references)
+                    data[Model._meta.pk.attname] = Model._meta.pk.to_python(obj.get('pk'))
                 except Exception as e:
-                    raise base.DeserializationError.WithData(e, d['model'], d.get('pk'), field_value)
-                if value == base.DEFER_FIELD:
-                    deferred_fields[field] = field_value
+                    raise base.DeserializationError.WithData(e, obj['model'], obj.get('pk'), None)
+
+            if self.ignorenonexistent:
+                if Model not in self.field_names_cache:
```

Comment:
Is not this condition always true?

Reply:
@likecodingloveproblems I didn't get why it's always true. Seems models in the object list can be repetitive. It's like the original code except for an added condition for **ignorenonexistent** because it's only applicable when it's true.